packages/@react-aria/focus/src/useFocusable.tsx

```diff
@@ -12,17 +12,53 @@
 
 import {FocusEvents, KeyboardEvents} from '@react-types/shared';
 import {mergeProps} from '@react-aria/utils';
-import {RefObject, useEffect} from 'react';
+import React, {HTMLAttributes, MutableRefObject, ReactNode, RefObject, useContext, useEffect} from 'react';
 import {useFocus, useKeyboard} from '@react-aria/interactions';
 
 interface FocusableProps extends FocusEvents, KeyboardEvents {
   isDisabled?: boolean,
   autoFocus?: boolean
 }
 
+interface FocusableContextProps extends HTMLAttributes<HTMLElement> {
+  children?: ReactNode,
+  ref?: MutableRefObject<HTMLElement>
+}
+
+export const FocusableContext = React.createContext<FocusableContextProps>(null);
+
+export function useFocsuableContext(ref: RefObject<HTMLElement>): FocusableContextProps {
+  let context = useContext(FocusableContext) || {} as FocusableContextProps;
+
+  useEffect(() => {
```

Comment:
have the same problem with ref as DOMPropsContext https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/%40react-aria/interactions/src/DOMPropsContext.ts#L36